### PR TITLE
View Specs and Resolving N+1 Problem ⬆️ 

### PR DESCRIPTION
**Pull request by demesameneshoa**
### Fixing the `N+1 Problem`

- [x] The `N+1 Problem` on Posts#index before fixing it
before

- [x] The `N+1 Problem` on Posts#index after fixing it
after
### Integration Specs Using `capybara` on the below views
- [x] `Users Index` View
- [x] `Users Show` View
- [x] `Posts Index` View
- [x] `Posts Show` View

### CODE REVIEW

File: spec/views/user_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    # I can click on a user's name to go to their show page.
+    it 'it should redirects to that users show page' do
+      # ? Factory Bot creates 5 users, so we need to take one of them to test. since all o them share
+      # ? the same data (e.g. name).
+      user = create(:user) # ! so just create a brand new one otherwise we get 'ambiguous match' error.
+      visit users_path
+      click_link user.name
+      expect(page).to have_current_path(user_path(user))
+    end
+  end
```
Comment (zieeco) [inline spec/views/user_index_spec.rb]: - Your test case examples fail. Please, can you reevaluate it?

image

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @Team,






Great job so far! 👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- Fix n+1 problem ✔️
- Nice design. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._




## Optional suggestions

- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/views/user_index_spec.rb (hunk 1 above)
Comment (demesameneshoa) [reply spec/views/user_index_spec.rb]: Resolved

**Review by Efiamotu-1 — APPROVED** [review]
Hi @Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# STATUS: APPROVED 🕺🥳 

image

## Highlights :dart: 
* Correct Gitflow :heavy_check_mark: 
* Good commit messages :heavy_check_mark: 
* All linters passing :heavy_check_mark: 
* Good PR summary :heavy_check_mark: 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, you can now Merge since it is Approved ✅, However if there are any conflicts please resolve them before merging.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**